python/pyarrow/scalar.pxi

```diff
@@ -639,12 +626,27 @@ cdef class StructScalar(Scalar, collections.abc.Mapping):
             else:
                 raise KeyError(key)
 
+    # TODO(kszucs): consider to remove __iter__, keys and values
```

Comment:
@jorisvandenbossche we need to clean this up once we decide about the desirable API

Reply:
Reverted so we can find out a better solution.